oak_dice/src/utils.rs

```diff
@@ -29,3 +33,16 @@ pub fn cbor_encoded_bytes_to_vec(bytes: &[u8]) -> Result<Vec<u8>, String> {
         .map_err(|err| format!("failed to write bytes: {:?}", err))?;
     Ok(result)
 }
+
+/// Like the slice `copy_from_slice` method but does not panic if slices are not
+/// the same length. In the case of a shorter source slice, only overwrites
+/// the beginning of the destination slice. In the case of a longer source slice
+/// it throws an error.
+pub(crate) fn padded_copy_from_slice(dst: &mut [u8], src: &[u8]) -> Result<(), String> {
```

Comment:
optional nit: `padded` is misleading here since it implies to me that it is zero-padded, but the functions does not zero out the rest of the dst if it contains other data. I would prefer either adding logic to zero out the rest of dst, or using a different name.

Reply:
Good point, updating with logic to zero out the rest